### DISCUSSION TITLE
supbase client query multiple related tables

### DISCUSSION THREAD

**Discussion by jstansbe** [supabase/supabase#2258] [Questions] [answered]
Say I have some tables
  posts: [id] [category_id] [post_text]
  post_categories: [id] [category_title]
  post_tags: [id] [post_id] [tag_text]

I want to select posts along with their associated categories and tags using supabase client.

The db query would ideally return a data shape like:
```
[{
    id: 1231fsf,
    post_text: 'Post text here',
    categories: [{  id: "123qwe", category_title: "cat title 1" }],
   tags: [{ id: "qwe123", tag_text: "Tag 1" }, { id: "zxc678", tag_text: "Tag 2" }]
}]
```

**Comment by inian** [comment, marked as answer]
Hi 👋🏽 , if you set the foreign keys between the tables correctly, you can get almost exactly what you are looking for by using [resource embedding](https://supabase.io/docs/reference/javascript/select#query-foreign-tables)

untested code
```
supabase.from('posts').select('*, post_categories(*), post_tags(*)').eq('id', '1231fsf')
```

**Reply by jstansbe** [reply]
That did the trick! So awesome to be able to grab nested data like this.

**Reply by ichbinedgar** [reply]
However, how I can use the same table twice?
```
....
 "Users":owner_id (
    name
  ),
  "Users":resident_id (
    name
  )
  ...
  ```

**Reply by ichbinedgar** [reply]
ok, I've just figured it out.  The first part is the alias, so it can be used like this:

```
 owner:owner_id (
    name
  ),
 resident:resident_id(
    name
  ) 
```

**Comment by mrmartineau** [comment]
@jstansbe I think I'm trying to do the same thing. Can I ask if you are missing some information in the posts table when you defined it above?

```
posts: [id] [category_id] [post_text]
```
Should it also have had `[tags]`? and if so, what type of field would it be? a text array field? I guess the same type of field would be needed for the `[category_id]` column? Did you allow multiple values for each of these columns?
